File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.py
class Solution:
    def myAtoi(self, s: str) -> int:
        s = s.lstrip(' ')
        if s == '':
            return 0
        negative = s[0] == '-'
        if negative or s[0] == '+':
            s = s[1:]
        s = s.lstrip('0')
        digits = [str(i) for i in range(10)]
        i = 0
        while i < len(s) and s[i] in digits:
            i = i + 1
        s = s[:i]
        if s == '':
            return 0
        x = int(s)
        if negative:
            x = -x
        if x < -2**31:
            return -2**31
        elif x > 2**31 - 1:
            return 2**31 - 1  
        return x

File: 0008-string-to-integer-atoi/0008-string-to-integer-atoi.cpp
class Solution {
public:
    int myAtoi(string s) {
        int i=0, sgn = 1;
        long long res = 0;
        while(i<s.size() && s[i]==' ') i++;
        if(i==s.size()) return 0;
        if(s[i]=='-'){
            sgn = -1; i++;
        }else if(s[i]=='+')i++;
        while(i<s.size() && isdigit(s[i])){
            res = res*10 + (s[i]-'0');
            if(sgn*res > INT_MAX) return INT_MAX;
            if(sgn*res < INT_MIN) return INT_MIN;
            i++;
        }
        return (int)(sgn*res);
    }
};